sub-packages/bionemo-dnadl/src/bionemo/dnadl/tools/vcf.py

```diff
@@ -0,0 +1,92 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: LicenseRef-Apache2
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from pathlib import Path
+
+import pandas as pd
+import pysam
+
+from bionemo.dnadl.tools.genome_interval import GenomeInterval
+
+
+SampleId = str | int
+
+
+def _get_max_allele_index(allele_indices: tuple[int | None, int | None]) -> int:
+    """Get the maximum allele index from a tuple of allele indices.
+
+    `sample.allele_indices` is a tuple of two indices that associate the sample with alleles
+    listed in the list 'variant.alleles', where the zeroth entry is the REF allele. A sample
+    could have one or both entries being 0, indicating that one or both strands are REF at
+    the specified position. By taking max(sample.allele_indices) you ensure to always extract
+    an alternative allele if present or REF otherwise.
+
+    Sometimes, None is contained in sample.allele_indices. This is treated as if it refers
+    to the REF allele (ie 0)
+    """
+    adjusted_allele_indices = [0 if idx is None else idx for idx in allele_indices]
+    max_allele_ind = max(adjusted_allele_indices)
+    return max_allele_ind
+
+
+def _match_contig_name(chrm: str, available_contigs: set[int | str]) -> str:
+    """Let's try either the standardized version that i expect to be "chr22" or only the number string "22"."""
```

Comment:
My opinion: we should just fail in this case. It's too easy to end up with a reference that has the wrong amount of `N`s or something else stupid.

Reply:
Happy to do that but FYI internally we did this because hg19 and hg38 VCF files used different conventions https://gatk.broadinstitute.org/hc/en-us/articles/360035890951-Human-genome-reference-builds-GRCh38-or-hg38-b37-hg19 . We wanted it to work with both. Theoretically you could say that the user needs to  take care of this, but it becomes less user friendly if you get into situations where windows on top of TSSs come from Gencode which uses a specific standard.

Reply:
fine with me, per our call- add a warning if the seqids dont match between the vcf and reference.